### PR TITLE
[miio] add gosund plug & Vacuum 1C STYTJ01ZHM

### PR DESCRIPTION
**Pull request by marcelrv**
[miio] add gosund plug & Vacuum 1C STYTJ01ZHM

Add 2 miot devices
Vacuum 1C STYTJ01ZHM dreame.vacuum.mc1808
cuco goplug cuco.plug.cp1

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

### CODE REVIEW

File: bundles/org.openhab.binding.miio/README.md [hunk 1]
```diff
@@ -1131,6 +1143,45 @@ e.g. `smarthome:send actionCommand 'upd_timer["1498595904821", "on"]'` would ena
...
+| BrushLeftTime1   | String  | Side Cleaning Brush-Brush Left Time |
+| BrushLifeLevel1  | Number  | Side Cleaning Brush-Brush Life Level |
+| WorkMode         | Number  | clean-workmode                      |
+| Area             | String  | clean-area                          |
+| Timer            | String  | clean-timer                         |
+| Mode             | Number  | clean-mode                          |
+| TotalCleanTime   | String  | clean-total time                    |
+| TotalCleanTimes  | String  | clean-total times                   |
+| TotalCleanArea   | String  | clean-                              |
+| CleanLogStartTime | String  | clean-                              |
```
Comment (cpmeister) [inline bundles/org.openhab.binding.miio/README.md:1171]: description is missing for these.

File: bundles/org.openhab.binding.miio/README.md [hunk 2]
```diff
@@ -1131,6 +1143,45 @@ e.g. `smarthome:send actionCommand 'upd_timer["1498595904821", "on"]'` would ena
...
+| ButtonLed        | String  | clean-led                           |
+| TaskDone         | Number  | clean-task done                     |
+| LifeSieve        | String  | consumable-life-sieve               |
+| LifeBrushSide    | String  | consumable-life-brush-side          |
+| LifeBrushMain    | String  | consumable-life-brush-main          |
```
Comment (cpmeister) [inline bundles/org.openhab.binding.miio/README.md:1176]: Are these supposed to be percentage values?

File: bundles/org.openhab.binding.miio/README.md [hunk 3]
```diff
@@ -1131,6 +1143,45 @@ e.g. `smarthome:send actionCommand 'upd_timer["1498595904821", "on"]'` would ena
...
+| FilterLifeLevel  | Number  | Filter-Filter Life Level            |
+| FilterLeftTime   | String  | Filter-Filter Left Time             |
+| BrushLeftTime1   | String  | Side Cleaning Brush-Brush Left Time |
+| BrushLifeLevel1  | Number  | Side Cleaning Brush-Brush Life Level |
+| WorkMode         | Number  | clean-workmode                      |
+| Area             | String  | clean-area                          |
+| Timer            | String  | clean-timer                         |
+| Mode             | Number  | clean-mode                          |
+| TotalCleanTime   | String  | clean-total time                    |
+| TotalCleanTimes  | String  | clean-total times                   |
```
Comment (cpmeister) [inline bundles/org.openhab.binding.miio/README.md:1169]: what is the difference between these fields? Can they be `Number:Time` instead?

**Comment by marcelrv** [comment]
Hallo @cpmeister 

This is a so called miot device, which is following the new protocol Xiaomi is trying to implement for their devices. (I think they are slowly moving all their devices to it)

For these devices they publish the definitions e.g. for the vacuum here.
http://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:vacuum:0000A006:dreame-mc1808:1

The OH database json file is a translation of that file, incl the data types. 
So, I don't really intend to change the data types e.g. string/number/etc and stick to the published definition unless we have confirmation from a user that owns such a device what it outputs.

In the end my intention is that these can be added directly by the user in the miio:unsupported handler ( https://github.com/openhab/openhab-addons/issues/7276 )

But before that, I want to give it a try for a few devices to see if it gives meaningful results.
e.g. I found indeed several properties miss the description. I fixed most of them, but seems I missed one.

note, the list of devices that support this (miot) protocol is already rather big (total list of Xiaomi devices >1500 by now)
Mi Air Purifier 3 (zhimi.airpurifier.ma4)
Mi Air Purifier 3/3H (zhimi.airpurifier.mb3)
were done with the initial PR, now 2 more.

One meter intelligent curtain (090615.curtain.1mcu01)
PTX Rolling curtain (090615.curtain.jldj03)
Intelligent 86 socket(WIFI) (090615.plug.plus01)
Intelligent Mobile Plug (WIFI) (090615.plug.plus02)
Piak intelligent one switch (090615.switch.piak01)
Piak intelligent two switch (090615.switch.piak02)
Piak intelligent three switch (090615.switch.piak03)
Ai-Thinker Smart Socket (180.plug.plug0001)
Smart Curtain Motor (666.curtain.gt01)
Airmate mobile floor heating (air.heater.wda14)
Nnuodu Intelligent Living Room Lamp (app.light.wynd1)
Feipujia Intelligent lamps and lanterns (asdds.light.wyfpj1)
AUTORAIL Curtain Motor (babai.curtain.at5810)
MTX Smart Curtain (babai.curtain.mtx850)
Babuy Smart Curtain (babai.curtain.bb82mj)
NanHoo Smart Curtain (babai.curtain.nh5810)
WiFi Wall Switch A1 (babai.switch.bb101s)
WiFi Wall Switch A3 (babai.switch.bb103s)
Smart Curtain Motor (babai.curtain.190812)
WiFi Wall Switch A1 (babai.switch.bb101s)
WiFi Wall Switch A2 (babai.switch.bb102s)
WiFi Wall Switch A3 (babai.switch.bb103s)
LOMEDIQI Intelligent automatic air purifier (bjdm.airfresh.test01)
Qingfly smart portable clothes dryer (coc.dryer.fdpsm)
Heatcold Heat Pump Thermostat (cubee.airrtc.th125t)
Heatcold UFH Thermostat (cubee.airrtc.th123w)
Heatcold FCU thermostat (cubee.airrtc.th125a)
Heatcold Heating Thermostat (cubee.airrtc.th123e)
Gosund Smart Plug (cuco.plug.cp1)
Gosund Socket (cuco.plug.cp2)
Cuco Smart Plug (wifi) Basic (cuco.plug.wp5)
Muzi Intelligent Light (cxds.light.wymz01)
NIOYE Intelligent lamp (ddk.light.wy0a01)
Deye dehumidifier (deye.derh.u20a3)
Dimi RO water purifier 650G (dmaker.waterpuri.600g)
DooyaSmartCurtain (dooya.curtain.m1)
Xiaomi Mijia Vacuum-Mop 1C (dreame.vacuum.mc1808)
A300 Splendid Laser Projector (fengmi.projector.l166)
Mi Laser Projector (fengmi.projector.l185)
Peak meter projector Vogue (fengmi.projector.m135f)
Fantong Intelligent Lamp (ftds.light.wyft1)
Gomanni Scene two color ceiling lamp (gmn.light.wy0w01)
MiWhole Air Purifier Mix (hanyi.airpurifier.kj550)
Langli intelligent LED ceiling lamp (hcznjj.light.wyll01)
Smart electric blanket (hddz.blanket.zndrt)
Hadaman intelligent LED ceiling lamp (hdm.light.wy08a)
HUAYI HUIZUO ARIES For Bedroom (huayi.light.ari013)
HUAYI HUIZUO ARIES For Living Room (huayi.light.aries)
HUAYI HUIZUO Fan Light (huayi.light.fanwy)
HUAYI HUIZUO PEGASUS For Living Room (huayi.light.peg091)
HUAYI HUIZUO PEGASUS For Bedroom (huayi.light.peg093)
HUAYI HUIZUO PISCES For Bedroom (huayi.light.pis123)
HUAYI HUIZUO PISCES For Living Room (huayi.light.pisces)
HUAYI HUIZUO TAURUS For Bedroom (huayi.light.tau023)
HUAYI HUIZUO TAURUS For Living Room (huayi.light.taurus)
HUAYI HUIZUO VIRGO For Bedroom (huayi.light.vir063)
HUAYI HUIZUO VIRGO For Living Room (huayi.light.virgo)
HUAYI HUIZUO Ceiling Light (huayi.light.wy)
Lifan Nordic Intelligent Lamp (hyzm.light.wymy1)
Danfino Intelligent Roof Suction Lamp (jsc.light.wydfn1)
Scene Ble mesh light (jsc.light.wyls02)
Yantai Intelligent Lamp (jsc.light.wyyt01)
Xiangyu intelligent LED ceiling lamp (jxgc.light.wyxy1)
Jya Convection Heater (jya.heater.sha1)
NVC LED Smart Ceiling Lamp (leishi.light.eps112)
Scene Ble mesh light A (lejia.light.33301)
Scene mesh color temperature lamp C (lejia.light.wy02)
Scene Color Light WIFI X (lemesh.light.wy)
Scene mesh color temperature lamp (lemesh.light.wy0c02)
Scene mesh color temperature lamp D (lemesh.light.wy0c03)
Scene mesh breaker (lemesh.switch.sw0a01)
Scene Curtain WIFI X (leshi.curtain.v0001)
Smart Switch WIFI X (leshi.switch.v0001)
Switch Breaker WIFI X (leshi.switch.v002)
Linptech Smart Wall Switch(1 Key) (linp.switch.q31)
Linptech Smart Wall Switch(2 Keys) (linp.switch.q32)
Linptech Smart Wall Switch(3 Keys) (linp.switch.q33)
Shengxin Intelligent lamp (lsds.light.wy0a01)
Lige lighting (ltdzsw.light.wylg1)
Xiaomiyoupin Curtain Controller (Wi-Fi) (lumi.curtain.hagl05)
Mi Smart Home Hub (lumi.gateway.mgl03)
Mi Light Detection Sensor (lumi.sen_ill.mgl01)
Aqara Smart Smoke Detector(NB-IoT) (lumi.sensor_smoke.acn01)
Smart Low Voltage Blanket Double (miaomiaoce.blanket.d02)
Smart Low Voltage Blanket Single (miaomiaoce.blanket.s02)
Cat and dog water fountain (mmgg.pet_waterer.s1)
MR.BOND (mrbond.airer.m2)
MR.BOND (mrbond.airer.m2pro)
MR.BOND Smart Curtain (mrbond.curtain.rac03)
NEW WIDETECH Internet Dehumidifier 12L (nwt.derh.312en)
NWT Internet Dehumidifier 30L (nwt.derh.330ef)
Ougenuo Intelligent lamp (ougn.light.wy0a01)
Zhirui Ceiling Lamp Nordic 40W (philips.light.mceilm)
Zhirui Ceiling Lamp Nordic 28W (philips.light.mceils)
Zhirui Ceiling Lamp Nordic 80W (philips.light.mceil)
Puzhuoer Smart Light (pze.light.wy0a01)
8H Milan smart electric bed (qushui.bed.001)
RBE wisdow living room lamp (rbe.light.wy0a01)
M-YF6002A Smart Bathroom Heater (roome.bhf_light.yf6004)
Petoneer AirMaster (skyrc.airpurifier.pur)
Chanitex Water-heated mattress CXM-A1 (smith.blanket.cxma1)
Shengteng Lighting Ceiling Light (stds.light.wyst01)
Shengteng Ceiling Light (stds.light.wyst02)
Bijia Fresh Air Ventilator (szdy.airfresh.n80)
DanceLight I wisdom kanLamp (tsd.light.tsl001)
Viomi Smart Bath Heater PRO (viomi.bhf_light.v3)
Viomi Smart IH Rice Cooker 3L (viomi.cooker.v4)
Viomi Kick-Line heater Pro (viomi.heater.vxtj01)
WinTom Curtain (wintom.curtain.230xm)
Ocooker hood (xckj.hood.z9001)
Ocooker french refrigerator with four doors (xckj.fridge.448ws4)
Ocooker dishwasher (xckj.dishwasher.idw01)
Ocooker smart refrigerator for kitchen (xckj.fridge.151wg1)
Romne Intelligent lamp (xgds.light.wy0a01)
Mi Smart Air Conditioner A (1HP / Inverter / China Energy Label Level 1) (xiaomi.aircondition.mc1)
Mi Smart Air Conditioner A (1.5HP / Inverter / China Energy Label Level 1) (xiaomi.aircondition.mc2)
Mi Smart Air Conditioner A (1HP / Inverter / China Energy Label Level <1) (xiaomi.aircondition.mc4)
Mi Smart Air Conditioner A (1.5HP / Inverter / China Energy Label Level <1) (xiaomi.aircondition.mc5)
Mi Smart Vertical Air Conditioner A (2HP / Inverter / China Energy Label Level <1) (xiaomi.aircondition.mc6)
Mi Smart Vertical Air Conditioner A (3HP / Inverter / China Energy Label Level <1) (xiaomi.aircondition.mc7)
Letsleep one click smart aqua-heat mattress (xinyue.blanket.n1918)
Xi Zhi Xi Intelligent Lamp (xzx.light.wyxzx1)
Yeelight Smart Curtain Controller (yeelink.curtain.ctmt1)
Yeelight Mesh LED Downlight (yeelink.light.dnlight2)
Yeelight Smart Ceiling Fan (yeelink.light.fancl1)
Yeelight Mesh LED Bulb (yeelink.light.meshbulb1)
Yeelight Mesh LED Bulb (yeelink.light.meshbulb2)
Yeelight Mesh LED Spotlight (yeelink.light.spot1)
Mi Smart Bluetooth Mesh LED Downlight (yeelink.light.light3)
Mi Smart Bluetooth Mesh LED Bulb (yeelink.light.mbulb3)
Feihong lighting (yhzm.light.wyfh01)
Leiyuan Intelligent Living Room Lamp (ylzm.light.wyly01)
EMG Top halo ceiling lamp (ymj.light.wyymj1)
Mi Air Purifier 3 (zhimi.airpurifier.ma4)
Mi Air Purifier 3/3H (zhimi.airpurifier.mb3)
Mi Air Purifier Pro H (zhimi.airpurifier.va1)
Mi Smart Space Heater S (zhimi.heater.ma2)
Smartmi Smart Fan (zhimi.heater.na1)
Smartmi Smart Convector Heater 1S (zhimi.heater.za2)
Smartmi Smart Convector Heater 1S (zhimi.heater.zb1)
Smartmi Evaporative Humidifier 2 (zhimi.humidifier.ca4)
ZSZM Intelligent ceiling lamp (zszm.light.wy0a03

File: bundles/org.openhab.binding.miio/README.md (hunk 2 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.miio/README.md:1176]: see the long comment below

File: bundles/org.openhab.binding.miio/README.md (hunk 3 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.miio/README.md:1169]: To be honest, I don't know. see the long comment below

File: bundles/org.openhab.binding.miio/README.md (hunk 1 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.miio/README.md:1171]: yep. fixed

File: bundles/org.openhab.binding.miio/README.md (hunk 3 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.miio/README.md:1169]: But, given the names, I expect that the one is the (clock) time of the cleaning and the other the amount of times the machine has cleaned. Similar counts are for the other Xiaomi vacuum cleaners available.

btw, it is still on my wish list #7276 to have the binding deal with `Number:Time` sort of notation. I expect it is not a huge effort, but I have not looked deeper into it what would be the detailed impact to the updating process and the possible impact in handling of the commands

**Pull request merged**